tests/mocks/MockFileSystem.hh

```diff
@@ -130,13 +130,13 @@ public:
             return false;
 
         hNode.key() = sNewPath;
-        m_mFileContents.insert(std::move(hNode));
+        (void)m_mFileContents.insert(std::move(hNode));
```

Comment:
It'll say "avoid unnamed objects with custom construction and destruction (RAII)".

Even if an RAII object does not have a `[[nodiscard]]` there shouldn't be an unnamed RAII object. The standard sanctioned why to suppress it is to cast to `void`. I'm not sure if a cast to void will trigger "Don't use C-Style casts" when the type profile is run, if it does